Source/DataAccessLayer.SqlClient/StoredProcedureProvider.cst

```diff
@@ -179,22 +179,23 @@ namespace <%=DALNameSpace%>.SqlClient
 		/// Get the command wrapper, either from the stored procedures or from the embedded queries.
 		/// </summary>
 		/// <param name="database">The database instance.</param>
+		/// <param name="defaultCommandTimeout">the timeout of the sp.</param>
 		/// <param name="commandName">the name of the sp.</param>
 		/// <param name="useStoredProcedure">Indicates if we want stored procedure.</param>
 		/// <returns></returns>
-		public static DbCommand GetCommandWrapper(Database database, string commandName, bool useStoredProcedure)
+		public static DbCommand GetCommandWrapper(Database database, int defaultCommandTimeout, string commandName, bool useStoredProcedure)
 		{
 			if (useStoredProcedure)
 			{
 				DbCommand command = database.GetStoredProcCommand(commandName);
-                command.CommandTimeout = DataRepository.Provider.DefaultCommandTimeout;
+                command.CommandTimeout = defaultCommandTimeout; //DataRepository.Provider.DefaultCommandTimeout;

```

Comment:
Should we make default command timeout the last argument and an optional and fall back to the default command timeout?

Reply:
We can't make it optional, optional arguments are a more recent invention in .net and this code base is supposed to be able to support older versions of .net.
I can not think of any benefit that would be gained by falling back to the default in this instance.  I feel the entire call into this method should be re-written, but that was a bigger project then I wanted to tackle at the moment.

Reply:
Ok, should we remove this commented out code then? I'm wondering if this will break any Code that might of been relying on the default provider timeout value.

Reply:
I looked through the code back when I did it and could not see any way it would break it.  However, in the off chance it did, that is why I left the code in there commented out.  I figured on the next time in there if everything has been good for a year I would rip out the comment.
That seems to follow the pattern used throughout the rest of the code base.